Add drf

jnation3406 opened the pull request:
Parts of this PR:

- Switch to using DRF
- Rename observatory->enclosure and updated->modified in the model
- Connect to obs portal for oauth authentication of posting data
- Connect to configdb for validation of site/enclosure/telescope
- Move api to /api and add frontend for viewing at /

The oauth authentication is currently just HTTPBasicAuth here, but I have a branch that supports getting and using a bearer token for auth. We are planning to revisit the auth model for all the OCS applications in a later sprint so that stuff may change.

timbeccue commented on README.md:
```diff
@@ -23,12 +26,19 @@ This project is configured using environment variables.
 | `DB_USER`            | Database Username, set this when using PostgreSQL                                  | _empty string_               |
 | `DB_PASS`            | Database Password, set this when using PostgreSQL                                  | _empty string_               |
 | `DB_PORT`            | Database Port, set this when using PostgreSQL                                      | `5432`                       |
+| `OAUTH_CLIENT_ID`    | OAuth authentication client id (found in observation portal admin for the app)     | ``                           |
```
Not sure who the target audience is for the readme, but it's not clear to me where I would find this value. Is it defined when creating the oauth server somewhere else? If it's something a new user trying to set things up (me) should be able to figure out, then additional details might be helpful.

jnation3406 replied:
The oauth client id and secret and autogenerated for you when you create an oauth "application" on the server. We should probably describe this better in overall project docs and reference from here...

eheinrich replied:
Agreed, when we work out how we want to do oauth across the projects, I think that would be a good time to create some documentation on it.

timbeccue reviewed (approved):
Looks fine to me.

eheinrich reviewed (commented):
The tests fail for me, and I get a 500 when I navigate to `/api/`, until I run `python manage.py collectstatic`. The index page works however. Is there a way to serve static files for local development without having to run collectstatic first?

jnation3406 commented on schedule/tests.py:
```diff
@@ -44,4 +45,62 @@ def test_downtime_with_end_before_start_not_allowed(self):
+        self.assertIn('End time must be after start time', str(response.content))
+
+
+class TestDowntimeSerializer(TestCase):
```
good idea, I'll add it

jnation3406 commented on schedule/templates/downtime_list.html:
```diff
@@ -0,0 +1,68 @@
+  <body>
+    <nav class="navbar navbar-expand-lg navbar-light navbar-bg head">
+      <div class="container ">
+        <span class=navbar-brand><img src={{ logo_url }} class="img-fluid"/></span>
```
Yeah I guess that seems like a good idea

jnation3406 commented:
hmm the tests appear to fail if you don't have configdb set, even though it is mocked. I think it still needs a http:// in it or requests breaks even with the mocking. I'll add a test_settings file like we do with the obs portal to set some sensible values for testing...

jnation3406 commented:
Also, I think if you set DEBUG to True, then you don't need to run collectstatic first.

jnation3406 commented on schedule/views.py:
```diff
@@ -1,25 +1,31 @@
+            kwargs['data'] = {
+                'ends_after': timezone.now()
+            }
+        return kwargs
```
weird codacy catches these things sometimes but misses them others.

jnation3406 commented:
I addressed all the comments. Couldn't find a way to test the admin pages without running collectstatic first, so I added that to the readme on how to run the tests.

eheinrich reviewed (approved):
Cool, looks good!
